frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -144,89 +151,147 @@ interface Interval {
   right: number
 }
 
-type IntervalData = "int64" | "uint64" | "float64" | "datetime64[ns]"
-type IntervalClosed = "left" | "right" | "both" | "neither"
-type IntervalType = `interval[${IntervalData}, ${IntervalClosed}]`
-
-/** Formats an interval index. */
-function formatIntervalType(data: StructRow, typeName: IntervalType): string {
-  const match = typeName.match(/interval\[(.+), (both|left|right|neither)\]/)
-  if (match === null) {
-    throw new Error(`Invalid interval type: ${typeName}`)
-  }
-  const [, subtype, closed] = match
-  return formatInterval(data, subtype, closed as IntervalClosed)
-}
-
-function formatInterval(
-  data: StructRow,
-  subtype: string,
-  closed: IntervalClosed
-): string {
-  const interval = data.toJSON() as Interval
-
-  const leftBracket = closed === "both" || closed === "left" ? "[" : "("
-  const rightBracket = closed === "both" || closed === "right" ? "]" : ")"
-  const leftInterval = format(interval.left, {
-    pandas_type: subtype,
-    numpy_type: subtype,
-  })
-  const rightInterval = format(interval.right, {
-    pandas_type: subtype,
-    numpy_type: subtype,
-  })
-
-  return `${leftBracket + leftInterval}, ${rightInterval + rightBracket}`
-}
-
 /**
  * Adjusts a time value to seconds based on the unit information in the field.
  *
  * The unit numbers are specified here:
  * https://github.com/apache/arrow/blob/3ab246f374c17a216d86edcfff7ff416b3cff803/js/src/enum.ts#L95
+ *
+ * @param timestamp The timestamp to convert.
+ * @param unit The unit of the timestamp. 0 is seconds, 1 is milliseconds, 2 is microseconds, 3 is nanoseconds.
+ * @returns The timestamp in seconds.
  */
-export function convertToSeconds(
-  value: number | bigint,
+function convertTimestampToSeconds(
+  timestamp: number | bigint,
   unit: number
```

Comment:
is it possible to use the `TimeUnit` type here that is used in the function body to compare `unit` against?

Reply:
Yep, changed it 👍